### PR TITLE
[VC-41] - Added Migration and Relationships between tables

### PR DESCRIPTION
**Pull request by BoshkaS**

### CODE REVIEW

File: VolunteerApp/Data/VolunteerContext.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        private IDbConnection DbConnection { get; }
+
+        private readonly IConfiguration _configuration;
+        public VolunteerContext (DbContextOptions<VolunteerContext> options , IConfiguration configuration)
+            : base(options)
+        {
+            this._configuration = configuration;
+            DbConnection = new SqlConnection(this._configuration.GetConnectionString("DefaultConnection"));
+            //Database.EnsureCreated();
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Data/VolunteerContext.cs:18]: Please remove

Code is always better than comments. So always avoid comments where code can be made self explanatory.

File: VolunteerApp/Data/VolunteerContext.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public class VolunteerContext : DbContext
+    {
+        private IDbConnection DbConnection { get; }
+
+        private readonly IConfiguration _configuration;
+        public VolunteerContext (DbContextOptions<VolunteerContext> options , IConfiguration configuration)
+            : base(options)
+        {
+            this._configuration = configuration;
+            DbConnection = new SqlConnection(this._configuration.GetConnectionString("DefaultConnection"));
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Data/VolunteerContext.cs:17]: Rename connection string;

For Exm: VolunteerDbConnectionString

File: VolunteerApp/Data/VolunteerContext.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            //Database.EnsureCreated();
+        }
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Data/VolunteerContext.cs:19]: An empty line is omitted

File: VolunteerApp/Data/VolunteerContext.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+                optionsBuilder.UseSqlServer(DbConnection.ToString());
+            }
+        }
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Data/VolunteerContext.cs:27]: An empty line before method is omitted

File: VolunteerApp/Data/VolunteerContext.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        private readonly IConfiguration _configuration;
+        public VolunteerContext (DbContextOptions<VolunteerContext> options , IConfiguration configuration)
+            : base(options)
+        {
+            this._configuration = configuration;
+            DbConnection = new SqlConnection(this._configuration.GetConnectionString("DefaultConnection"));
+            //Database.EnsureCreated();
+        }
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Data/VolunteerContext.cs:20]: An empty line before method is omitted

File: VolunteerApp/Models/Category_Request.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿namespace VolunteerApp.Models
+{
+    public class Category_Request
+    {
+        public int CategoryId { get; set; }
+        public Category Category { get; set; }
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Models/Category_Request.cs:6]: An empty line is omitted

File: VolunteerApp/Models/Category.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+
+namespace VolunteerApp.Models
+{
+    public class Category
+    {
+        [Key]
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Models/Category.cs:8]: https://www.entityframeworktutorial.net/efcore/fluent-api-in-entity-framework-core.aspx

File: VolunteerApp/Models/Guest.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace VolunteerApp.Models
+{
+    public class Guest
+    {
+        [Key]
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Models/Guest.cs:7]: https://www.entityframeworktutorial.net/efcore/fluent-api-in-entity-framework-core.aspx

File: VolunteerApp/Program.cs [hunk 9]
```diff
@@ -1,27 +1,39 @@
...
-
-// Configure the HTTP request pipeline.
-if (!app.Environment.IsDevelopment())
+namespace VolunteerApp
 {
-    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
-    app.UseHsts();
+    class Program
+    {
+        //static void Main(string[] args)
```
Comment (IrynaPanchuk97) [inline VolunteerApp/Program.cs]: Please remove comments

**Review by IrynaPanchuk97 — CHANGES_REQUESTED** [review]
https://www.entityframeworktutorial.net/efcore/fluent-api-in-entity-framework-core.aspx

**Review by Voridan — APPROVED** [review]

**Review by Ostap-04 — APPROVED** [review]

**Review by ykarpa — APPROVED** [review]

**Review by thevictoriax — APPROVED** [review]

File: VolunteerApp/Data/VolunteerContext.cs (hunk 2 above)
Comment (BoshkaS) [reply VolunteerApp/Data/VolunteerContext.cs:17]: fix

File: VolunteerApp/Data/VolunteerContext.cs (hunk 4 above)
Comment (BoshkaS) [reply VolunteerApp/Data/VolunteerContext.cs:27]: fixed

File: VolunteerApp/Data/VolunteerContext.cs (hunk 5 above)
Comment (BoshkaS) [reply VolunteerApp/Data/VolunteerContext.cs:20]: fixed

File: VolunteerApp/Models/Category_Request.cs (hunk 6 above)
Comment (BoshkaS) [reply VolunteerApp/Models/Category_Request.cs:6]: fixed

File: VolunteerApp/Models/Category.cs (hunk 7 above)
Comment (BoshkaS) [reply VolunteerApp/Models/Category.cs:8]: fixed

File: VolunteerApp/Models/Guest.cs (hunk 8 above)
Comment (BoshkaS) [reply VolunteerApp/Models/Guest.cs:7]: fixed

File: VolunteerApp/Program.cs (hunk 9 above)
Comment (BoshkaS) [reply VolunteerApp/Program.cs]: fixed

**Review by IrynaPanchuk97 — APPROVED** [review]

**Pull request merged**